openfl/federated/plan/plan.py

```diff
@@ -477,8 +481,9 @@ def get_task_runner(self, data_loader):
         if self.runner_ is None:
             self.runner_ = Plan.build(**defaults)
 
-        # Define task dependencies after taskrunner has been initialized
-        self.runner_.initialize_tensorkeys_for_functions()
+        if self.config["aggregator"][SETTINGS].get("mode", "learning") == "learning":
+            # Define task dependencies after taskrunner has been initialized
+            self.runner_.initialize_tensorkeys_for_functions()
```

Comment:
Can this be taken care within the initialize function? Please try not adding if-else unless absolutely required. This increases the (already high) cyclomatic complexity of the entire codebase.